Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -49,52 +51,214 @@ public FunctionManager(IEnvironment environment, ITraceFactory traceFactory)
+        }
+
+        private string GetFunctionSecretsFile(string functionName)
         {
-            foreach (var functionPath in FileSystemHelpers.GetDirectories(_environment.WebRootPath))
+            return Path.Combine(_environment.DataPath, Constants.Functions, Constants.Secrets, $"{functionName}.json");
+        }
+
+        // Logic for this function is copied from here
+        // https://github.com/Azure/azure-webjobs-sdk-script/blob/e0a783e882dd8680bf23e3c8818fb9638071c21d/src/WebJobs.Script/Config/ScriptHost.cs#L113-L150
+        private string GetFunctionScriptPath(string functionName, JObject functionInfo)
+        {
+            var functionPath = GetFunctionPath(functionName);
+            var functionFiles = FileSystemHelpers.GetFiles(functionPath, "*.*", SearchOption.TopDirectoryOnly)
+                .Where(p => Path.GetFileName(p).ToLowerInvariant() != "function.json").ToArray();
+
+            if (functionFiles.Length == 0)
             {
-                var functionJson = Path.Combine(functionPath, FunctionJsonFile);
-                if (FileSystemHelpers.FileExists(functionJson))
+                return functionPath;
+            }
+            else if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
+            }
+            else
+            {
+                // if there is a "run" file, that file is primary
+                string functionPrimary = null;
+                functionPrimary = functionFiles.FirstOrDefault(p => Path.GetFileNameWithoutExtension(p).ToLowerInvariant() == "run");
+                if (string.IsNullOrEmpty(functionPrimary))
                 {
-                    yield return functionJson;
+                    // for Node, any index.js file is primary
+                    functionPrimary = functionFiles.FirstOrDefault(p => Path.GetFileName(p).ToLowerInvariant() == "index.js");
+                    if (string.IsNullOrEmpty(functionPrimary))
+                    {
+                        // finally, if there is an explicit primary file indicated
+                        // in config, use it
+                        JToken token = functionInfo["source"];
+                        if (token != null)
+                        {
+                            string sourceFileName = (string)token;
+                            functionPrimary = Path.Combine(functionPath, sourceFileName);
+                        }
+                    }
                 }
+
+                if (string.IsNullOrEmpty(functionPrimary))
+                {
+                    // TODO: should this be an error?
+                    return functionPath;
+                }
+                return functionPrimary;
             }
         }
+
+        private Uri FilePathToVfsUri(string filePath, bool isDirectory = false)
```

Comment:
I couldn't find any. I was looking at how the WebJobs does it, [but it seems to build the urls one by one](https://github.com/projectkudu/kudu/blob/ahmels-functions/Kudu.Core/Jobs/JobsManagerBase.cs#L427-L435).